library/clinical_rules.php

```diff
@@ -1154,79 +1220,100 @@ function test_rules_clinic($provider = '', $type = '', $dateTarget = '', $mode =
+                                    $temp_track_pass = 1;
                                 }
-                            }
 
-                            break;
-                        } else {
-                            // send to reminder results
-                            if ($mode != "report") {
-                                // place the actions into the reminder return array
-                                $actionArray = resolve_action_sql($rowRule['id'], $i);
-                                foreach ($actionArray as $action) {
-                                    $action_plus = $action;
-                                    $action_plus['due_status'] = $reminder_due;
-                                    $action_plus['pid'] = $rowPatient['pid'];
-                                    $action_plus['rule_id'] = $rowRule['id'];
-                                    $results = reminder_results_integrate($results, $action_plus);
+                                // send to reminder results
+                                if ($mode == "reminders-all") {
+                                    // place the completed actions into the reminder return array
+                                    $actionArray = resolve_action_sql($rowRule['id'], $i);
+                                    foreach ($actionArray as $action) {
+                                        $action_plus = $action;
+
+                                        // original line:
+                                        //$action_plus['due_status'] = "not_due";
+                                        // With this line, Reminder Details page shows only not due and past due
+
+                                        // HR: My replacement lines
+                                        // if passed during this pass, this is the status
+                                        if ($dateCounter == 1) {
+                                            // not_due doesn't cause any text to show. Labels come from Clinical Rules Reminder Due Options. Has not_due, so not sure why that does not work
+                                            $reminder_status = "Not Due";
+                                        } elseif ($dateCounter == 2) {
+                                            $reminder_status = "soon_due";
+                                        } else { // $dateCounter == 3
+                                            $reminder_status = "due";
+                                        }
+
+                                        // HR: will leave next line alone. Not sure what implications of changing this are here. This was changed above for
+                                        //    if ((count($targetGroups) == 1) || ($mode == "report")) {
+                                        // case
+
+                                        $action_plus['due_status'] = "not_due";
+                                        //$action_plus['due_status'] = $reminder_status;
```

Comment:
@hanksterr7 , Doesn't above line negate the codes changes in lines 1271-1289 ?

Reply:
I will review. I agree I seem to be setting status in two places, somewhat redundantly, in both the count($targetGroups) = 1 and count($targetGroups) > 1 sections

Reply:
@bradymiller , I need your help understanding this code, which is in lines 1068-1104 of current released version of clinical_rules.php, and also 1191 - 1227. (I don't know how to do the nice trick of embedding code in green box with line numbers)
`                    if ($passTarget) {
                        // increment pass target counter
                        $pass_target++;
                        // If report itemization is turned on, then record the "passed" item and set the flag
                        if (!empty($GLOBALS['report_itemizing_temp_flag_and_id'])) {
                            insertItemReportTracker($GLOBALS['report_itemizing_temp_flag_and_id'], $GLOBALS['report_itemized_test_id_iterator'], 1, $rowPatient['pid']);
                            $temp_track_pass = 1;
                        }

                        // send to reminder results
                        if ($mode == "reminders-all") {
                            // place the completed actions into the reminder return array
                            $actionArray = resolve_action_sql($rowRule['id'], '1');
                            foreach ($actionArray as $action) {
                                $action_plus = $action;
                                $action_plus['due_status'] = "not_due";
                                $action_plus['pid'] = $rowPatient['pid'];
                                $action_plus['rule_id'] = $rowRule['id'];
                                $results = reminder_results_integrate($results, $action_plus);
                            }
                        }

                        break;
                    } else {
                        // send to reminder results
                        if ($mode != "report") {
                            // place the uncompleted actions into the reminder return array
                            $actionArray = resolve_action_sql($rowRule['id'], '1');
                            foreach ($actionArray as $action) {
                                $action_plus = $action;
                                $action_plus['due_status'] = $reminder_due;
                                $action_plus['pid'] = $rowPatient['pid'];
                                $action_plus['rule_id'] = $rowRule['id'];
                                $results = reminder_results_integrate($results, $action_plus);
                            }
                        }
                    }`
The above code is called for each pass through 
    foreach ($target_dates as $dateFocus) {
so it is called 3 times, each with a different $dateFocus value (and correspondingly, a $dateCounter value of 1, 2 or 3)
If the target passes, and $mode == "reminders-all", the code is setting $action_plus['due_status']  to 'not_due' for each of the actions in $actionArray.
If the target does not pass, and $mode != "report", then  $action_plus['due_status'] is getting set to the current "failure" status, as contained in $reminder_due (which is set a bit above this section), for each of the actions in $actionArray

This logic is not correct. If the $passTarget is true on pass 1, status should be not_due. If not true on pass 1, but true on pass 2, status should be soon_due. If not on passes 1 or 2, but true on pass 3, status should be due. Otherwise, status should be past_due. The code is instead setting the status to not_due regardless of the pass in which the target is satisfied.

My code fixes this, setting $action_plus['due_status'] to an appropriate value, regardless of whether the target succeeded or not. I did this by using both $reminder_due and $reminder_status.

I expect this was somewhat redundant in my code, and there is a cleaner way to do this.

But the part I don't understand is why the code is doing
    $results = reminder_results_integrate($results, $action_plus);
on each of the 3 passes, regardless of whether the target was satisfied or not. I didn't want to alter that behavior, but I didn't understand the purpose. If I did, I believe I could have found a cleaner way to do all of this.

I appreciate your suggestions for a better approach
Thanks

Reply:
yep, looks like I messed up on this logic. Posted a fix here:
https://github.com/openemr/openemr/pull/7211

basically need to ensure `$dateCounter == 1` before treating the target (and also the reporting code) as an official pass (ie. Not Due).

Reply:
@bradymiller what does "an official pass" mean? Is there a difference between "not due" and "due soon", for some reporting purpose, such that you want to increment $pass_target only if the status is "not due" and not if it is "due soon"?

Reply:
hi @hanksterr7 , sorry for the delay on replying to this. Planning to research this more over next couple days and will get back to you.